samples/templates/express-direct-auth/web-server/middlewares/authTransaction.js

```diff
@@ -5,15 +5,20 @@ function uniqueId() {
 }
 
 module.exports = function authTransaction(req, res, next) {
-  let transactionId;
-  if (req.query.transactionId) {
-    transactionId = req.query.transactionId;
-  } else if (req.query.state) {
-    transactionId = req.query.state;
-  } else {
-    transactionId = uniqueId();
-  }
-  req.transactionId = transactionId;
+  const { 
+    transactionId: transactionIdFromQuery, 
+    state: stateFromQuery, 
+  } = req.query;
+  const { 
```

Comment:
if you are just reading transactionId from session, then a new tab will use the same transactionId. This is what the transactionId is trying to prevent.

Reply:
So if I understand it correctly, we want to prevent concurrency issue for flow that happens in both server and browser sides, like SIW, social login, by using query param. But for flows that fully owned by the server side, it's safe to use session. 

If that's the case, I think we can add more comment/doc/option for the hybrid scenarios. The change here actually fixes the issue when no query param in the url, a new transaction will always be created.